lib/core/include/key_value_proxy.hpp

```diff
@@ -0,0 +1,540 @@
+            /// \returns bool
+            /// \retval true if the value for this handle is equal to the specified string; otherwise, false.
+            ///
+            /// \since 4.2.8
+            friend auto operator==(const handle& _h, const std::string& _s) -> bool
+            {
+                return _s == _h.val_;
+            }
+
+        private:
+            /// \var value_type val_
+            ///
+            /// \brief The value for the key-value pair managed by this handle
+            ///
+            /// \since 4.2.8
+            kvp* kvp_;
+
+            /// \var size_type index_
+            ///
+            /// \brief The index of the key-value pair in the kvp array
+            ///
+            /// \since 4.2.8
+            size_type index_;
+
+            /// \var value_type val_
+            ///
+            /// \brief The value for the key-value pair managed by this handle
+            ///
+            /// \since 4.2.8
+            value_type val_;
+
+            /// \fn static size_type index_of(key_type _k, kvp& _kvp)
+            ///
+            /// \brief Returns index of the specified key in the kvp array.
+            ///
+            /// \returns size_type
+            /// \retval index of the specified key in the kvp if found; otherwise, -1.
+            ///
+            /// \since 4.2.8
+            static auto index_of(key_type _k, kvp& _kvp) -> size_type
+            {
+                for (size_type i = 0; i < _kvp.len; i++) {
+                    if (_k == _kvp.keyWord[i]) {
+                        return i;
+                    }
+                }
+                return -1;
+            }
+        };
+
+        /// \class iterator
+        ///
+        /// \brief Iterator for array of kvps managed by a key_value_proxy
+        ///
+        /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+        ///
+        /// \since 4.2.8
+        class iterator {
+        public:
+            // iterator_traits: https://en.cppreference.com/w/cpp/iterator/iterator_traits
+            using value_type        = handle;
+            using pointer           = const value_type*;
+            using reference         = value_type;
+            using difference_type   = size_type;
+            using iterator_category = std::forward_iterator_tag;
+
+            /// \fn iterator(kvp& _kvp)
+            ///
+            /// \brief Default constructor for iterator
+            ///
+            /// Initializes to invalid index and nullptr for kvp
+            ///
+            /// \since 4.2.8
+            iterator()
+                : index_{-1}
+                , kvp_{}
+            {
+            }
+
+            /// \fn iterator(kvp& _kvp)
+            ///
+            /// \brief Constructs iterator for array of kvps starting at index 0
+            ///
+            /// \see https://en.cppreference.com/w/cpp/iterator/iterator
+            ///
+            /// \since 4.2.8
+            iterator(kvp& _kvp)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'iterator' has a constructor with 1 argument that is not explicit.](https://app.codacy.com/app/trel/irods/pullRequest?prid=4508572)